src/gausskernel/storage/access/k2/session.cpp

```diff
@@ -199,12 +172,114 @@ sh::Response<>  TxnManager::CreateCollection(const std::string& collection_name,
             .writeIOPs = 0,
             .minNodes = 1   // K2 Http proxy hangs if minNodes = 0
         },
-        .retentionPeriod = sh::Duration(1h) * 90 * 24,  //TODO: get this from config or from param in
+        .retentionPeriod = cconf.getDurationMillis("retention_period", 1h*24*90),
         .heartbeatDeadline = sh::Duration(0),
         .deleted = false
     };
 
-    return CreateCollection(metadata, rangeEnds);
+    return createCollection(metadata, rangeEnds);
+}
+
+boost::future<sh::Response<sh::dto::SKVRecord>>
+TxnManager::read(sh::dto::SKVRecord& record) {
+    K2LOG_D(log::k2pg, "read: {}", record);
+    return beginTxn()
+        .then([this, &record](auto&& beginFut) mutable {
```

Comment:
Safe to capture record by reference here?

Reply:
same contract as the underlying library - you keep the reference around for the duration of call

Reply:
changed to use values instead of references